윤건수/programmers/LV2_42583/Solution.java

```diff
@@ -0,0 +1,63 @@
+package programmers.LV2_42583;
+
+import java.util.*;
+
+public class Solution {
+    public int solution(int bridgeLength, int maxWeight, int[] truckWeights) {
+
+        Stack<Truck> weightTruck = new Stack<>();
+        Queue<Truck> bridgeTruck = new LinkedList<>();
+        for(int i=truckWeights.length-1; i>=0; i--){
+            weightTruck.push(new Truck(bridgeLength, truckWeights[i]));
+        }
+
+        int bridgeWeight = 0;
+        int time = 0;
+        while(!weightTruck.isEmpty() || !bridgeTruck.isEmpty()){
+            time++;
+
+            int size = bridgeTruck.size();
+            for(int i=0; i<size; i++){
+                Truck truck = bridgeTruck.poll();
+                bridgeWeight -= truck.weight;
+                truck.move();
+                if(!truck.isArrive()){
+                    bridgeTruck.add(truck);
+                    bridgeWeight += truck.weight;
+                }
+            }
+
+            Truck newTruck = null;
+            if(!weightTruck.isEmpty()
+                    && (maxWeight >= (weightTruck.peek().weight + bridgeWeight))
+                    && (bridgeTruck.size() <= bridgeLength)
+            ){
+                newTruck = weightTruck.pop();
+                bridgeWeight += newTruck.weight;
+                bridgeTruck.add(newTruck);
+                newTruck.move();
+            }
```

Comment:
```suggestion
            if(!weightTruck.isEmpty()
                    && (maxWeight >= (weightTruck.peek().weight + bridgeWeight))
                    && (bridgeTruck.size() <= bridgeLength)
            ){
                Truck newTruck = weightTruck.pop();
                bridgeWeight += newTruck.weight;
                bridgeTruck.add(newTruck);
                newTruck.move();
            }
```

newTruck  이 사용되는 부분이 if 안쪽이니까 옮겨주면 좋을 것 같네요.

Reply:
문제가 잘 안풀려서 이조건 저 조건 실험하다가 아마 외부에 변수 선언부를 남겨둔것 같네요 ㅎㅎ 초기엔 아마 해당 변수로 if 문 밖에서 무언가 했던거 같은데 좋은지적 같으십니다 ㅎㅎ